src/adapter/src/catalog.rs

```diff
@@ -2085,6 +2087,74 @@ impl<S: Append> Catalog<S> {
         Ok(())
     }
 
+    /// Create and insert the per replica log sources and log views.
+    ///
+    /// The replica with replica_id must already be present in the catalog state.
+    /// This function will allocate the oid's. The GlobalIds are already pre-allocated.
+    /// Views can depend on introspection sources and on other views.
+    fn insert_per_replica_items(
+        &self,
+        state: &mut CatalogState,
+        persisted_logs: &ConcreteComputeInstanceReplicaLogging,
+        replica_id: u64,
```

Comment:
It's a bit scary to me that this takes a reference to `self` which has one copy of a `CatalogState` and also takes a reference to `state` which is a potentially different copy of a `CatalogState`. It doesn't look like it causes any issues as far as I can tell, but may make it easy to have some consistency issue between the two states.
Is there any way to implement this method in `CatalogState` itself to avoid this?